barretenberg/cpp/src/barretenberg/dsl/acir_proofs/c_bind.cpp

```diff
@@ -125,14 +127,16 @@ WASM_EXPORT void acir_get_proving_key(in_ptr acir_composer_ptr, uint8_t const* a
 WASM_EXPORT void acir_goblin_verify_accumulator(in_ptr acir_composer_ptr, uint8_t const* proof_buf, bool* result)
 {
     auto acir_composer = reinterpret_cast<acir_proofs::GoblinAcirComposer*>(*acir_composer_ptr);
-    auto proof = from_buffer<std::vector<uint8_t>>(proof_buf);
+    auto proof_data_buf = from_buffer<std::vector<uint8_t>>(proof_buf);
+    auto proof = from_buffer<std::vector<bb::fr>>(proof_data_buf);
     *result = acir_composer->verify_accumulator(proof);
 }
 
 WASM_EXPORT void acir_goblin_verify(in_ptr acir_composer_ptr, uint8_t const* proof_buf, bool* result)
 {
     auto acir_composer = reinterpret_cast<acir_proofs::GoblinAcirComposer*>(*acir_composer_ptr);
-    auto proof = from_buffer<std::vector<uint8_t>>(proof_buf);
+    auto proof_data_buf = from_buffer<std::vector<uint8_t>>(proof_buf);
+    auto proof = from_buffer<std::vector<bb::fr>>(proof_data_buf);
```

Comment:
We can't skip an intermediate step because to_heap_buffer does not call to_buffer<true> on the first call. If I can change that part of to_heap_buffer without breaking other things, then it might be fine, but just went with this to be safe.